shell/platform/android/android_surface_software.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/common/threads.h"
+#include "flutter/fml/platform/android/jni_weak_ref.h"
+#include "flutter/fml/platform/android/scoped_java_ref.h"
+#include "flutter/shell/platform/android/android_surface_software.h"
+#include "flutter/shell/platform/android/platform_view_android_jni.h"
+
+
+#include <memory>
+#include <vector>
+
+#include "flutter/fml/trace_event.h"
+#include "lib/ftl/logging.h"
+
+namespace shell {
+
+AndroidSurfaceSoftware::AndroidSurfaceSoftware()
+    : AndroidSurface() {
+  
+}
+
+AndroidSurfaceSoftware::~AndroidSurfaceSoftware() = default;
+
+bool AndroidSurfaceSoftware::IsValid() const {
+  return true;
+  return sk_surface_ != nullptr;
+}
+
+bool AndroidSurfaceSoftware::ResourceContextMakeCurrent() {
+  return false;
```

Comment:
I'd add a comment here saying that the resource context is never available on the software backend.